File: perl/examples/nth_word.pl
#!/usr/bin/perl -w
# written by andrewt@cse.unsw.edu.au as a 2041 lecture example
# Print the nth word on every line of input files/stdin
# output is piped through fmt to make reading easy

die "Usage: $0 <n> <files>\n" if !@ARGV;
$nth_word = shift @ARGV;
open F, "|fmt -w 40" or die "Can nor run fmt";
while ($line = <>) {
    chomp $line;
    @words = split(/ /, $line);
    print F "$words[$nth_word]\n" if $words[$nth_word];
}
close F;

File: python/examples/nth_word.py
#!/usr/bin/python
# written by andrewt@cse.unsw.edu.au as a 2041 lecture example
# Print the nth word on every line of input files/stdin
# output is piped through fmt to make reading easy

import fileinput, sys, subprocess

if len(sys.argv) < 2:
    print >>sys.stderr, "Usage: %s <n>" % sys.argv[0]
    sys.exit(1)
    
nth_word = int(sys.argv.pop(1))
p = subprocess.Popen(["fmt","-w","40"], stdin=subprocess.PIPE)
for line in fileinput.input():
    words = line.rstrip().split()
    if len(words) > nth_word:
        print >>p.stdin, words[nth_word]
p.stdin.close()
